src/ADNLPProblems/hs316.jl

```diff
@@ -0,0 +1,18 @@
+export hs316
+
+function hs378(; n::Int = default_nvar, type::Val{T} = Val(Float64), kwargs...) where {T}
+  A = T[-6.089, -17.164, -34.054, -5.914, -24.721, -14.986, -24.1, -10.708, -26.662, -22.179]
+  function f(x; A=A)
+      return sum(exp(x[i])*(A[i]+x[i]-log(sum(exp(x[j]) for j = 1:10))) for i = 1:10)
+  end
+  function c!(cx, x)
+      cx[1] = exp(x[1])+2*exp(x[2])+2*exp(x[3])+exp(x[6])+exp(x[10])-2
+      cx[2] = exp(x[4])+2*exp(x[5])+exp(x[6])+exp(x[7])-1
+      cx[3] = exp(x[3])+exp(x[7])+exp(x[8])+2*exp(x[9])+exp(x[10])-1
+      return cx
+  end
+  x0 = convert(Vector{T}, -2.3 .* ones(10))
```

Comment:
an alternative without convert is:
```suggestion
  x0 = -23 // 10 .* ones(T, 10))
```